python/paddle/v2/reader/decorator.py

```diff
@@ -323,3 +323,101 @@ def xreader():
                 yield sample
 
     return xreader
+
+
+def _buf2lines(buf, line_break="\n"):
+    # FIXME: line_break should be automatically configured.
+    lines = buf.split(line_break)
+    return lines[:-1], lines[-1]
+
+
+def pipe_reader(left_cmd,
+                parser,
+                bufsize=8192,
+                file_type="plain",
```

Comment:
Maybe we just need to support "plain", the user can decompress it outside of Paddle using pipe.

Reply:
Thought it may be inconvenient for users to decompress stream data in their parsers.

Reply:
I meant the user can decompress the data using shell commands, not in the parsers, e.g.:
```bash
hadoop fs -cat /path/to/some/file | gzip -d
```

Reply:
Well, this is simpler, but I'm considering the pipe size using `bash` is set by `ulimit`, when in cluster trainer, user may not have control over every node's `ulimit` configuration, but using python code can.

Reply:
I don't understand bash very well, but does the pipe just "block" if it's full, and probably gzip can decode in a stream fashion, and will consume the pipe buffer, so it will be unblocked.

Reply:
By default, pipes can block both producer and consumer:

> If a process attempts to read from an empty pipe, then read(2) will
       block until data is available.  If a process attempts to write to a
       full pipe (see below), then write(2) blocks until sufficient data has
       been read from the pipe to allow the write to complete. 

Well, my point is, use pipes in python code, can let users to define pipe buffer size which is critical to the reader performance.

Reply:
I see, ok.